### PR TITLE
Fix #5754 - Error with custom fields on getQuery from One2Many relationships

### LINKED ISSUE #5754: Error with custom fields on getQuery from One2Many relationships
#### Issue

If you call `getQuery` method on One2Many relationship using custom fields on `order_by` or `where` optional parameters you've got the following error:

```
Query Failed: SELECT id FROM tasks WHERE tasks.contact_id = '62f50b22-c255-ccc9-a2e2-5a871e7b989' AND tasks.deleted=0 AND tasks_cstm.tipo_tarea_c = 'tipo1' LIMIT 0,10: MySQL error 1054: Unknown column 'tasks_cstm.tipo_tarea_c' in 'where clause'
```

This is caused due missing JOIN with cstm table.

#### Expected Behavior

It must be possible to call `getQuery` method on One2Many relationship using custom fields on `order_by` or `where`. This is working correctly with Many2Many relationships.

#### Actual Behavior

See issue description.

#### Possible Fix

See #5755

#### Steps to Reproduce

1. Call `getQuery` method on One2Many relationship using custom fields on `order_by` or `where`.
2. Notice that you do not receive any result.
3. See a MySQL error on `suitecrm.log`.

#### Context

I'm trying to get related records of a bean filtering by a custom field.

#### Your Environment

* SuiteCRM Version used: 7.8.18 LTS
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Versión 66.0.3359.117 (Build oficial) (64 bits)
* Environment name and version (e.g. MySQL, PHP 7): PHP 7.0.27-0 and mysql-server 5.5
* Operating System and version (e.g Ubuntu 16.04): Debian 9

### PR DESCRIPTION
**Pull request by sanchezfauste**
If you call `getQuery` method on One2Many relationship using custom fields on `order_by` or `where` optional parameters you've got the following error:

```
Query Failed: SELECT id FROM tasks WHERE tasks.contact_id = '62f50b22-c255-ccc9-a2e2-5a871e7b989' AND tasks.deleted=0 AND tasks_cstm.tipo_tarea_c = 'tipo1' LIMIT 0,10: MySQL error 1054: Unknown column 'tasks_cstm.tipo_tarea_c' in 'where clause'
```

This is caused due missing JOIN with cstm table.

See #5754 

## Description of changes
If optional `order_by` or `where` parameter is specified on `getQuery` method of One2Many relationship do a JOIN with custom table of module (if has custom table).

## Motivation and Context

I'm trying to get related records of a bean filtering by a custom field.

## How To Test This

1. Call `getQuery` method on One2Many relationship using custom fields on `order_by` or `where`.
2. Notice that you do not receive any result.
3. See a MySQL error on `suitecrm.log`.
4. Apply the PR and repeat steps 1, 2 and 3.
5. See that is working now.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

File: data/Relationships/One2MBeanRelationship.php [hunk 1]
```diff
@@ -263,8 +269,8 @@ public function getQuery($link, $params = array())
                 return $query;
             } else {
                 return array(
-                    'select' => "SELECT {$this->def['rhs_table']}.id",
-                    'from' => "FROM {$this->def['rhs_table']}",
+                    'select' => "SELECT {$rhsTable}.id",
```
Comment (code-ph0y) [inline data/Relationships/One2MBeanRelationship.php]: Could you escape the values being entered into the query using the quote or quoted function.

Comment (sanchezfauste) [reply data/Relationships/One2MBeanRelationship.php]: Hi @code-ph0y , I have refactored the code of getQuery function using quote and quoted functions when needed. Do you suggest any more changes to improve this PR?

**Comment by cameronblaikie** [comment]
Assessed :+1:

**Comment by Mac-Rae** [comment]
Please rebase this pull request to the hotfix-7.10.x branch as version 7.8.x of SuiteCRM has reached its end of life and therefore will no longer be receiving updates. 

For more information about 7.8.x end of life please visit our [blog post](https://suitecrm.com/7-8-x-end-of-life/)

**Comment by sanchezfauste** [comment]
> Please rebase this pull request to the hotfix-7.10.x branch as version 7.8.x of SuiteCRM has reached its end of life and therefore will no longer be receiving updates.
> 
> For more information about 7.8.x end of life please visit our [blog post](https://suitecrm.com/7-8-x-end-of-life/)

Hi @Mac-Rae , this PR is already merged on hotfix-7.10.x (see #7218), so if 7.8.x is no longer maintained we can close this PR.

**Pull request closed without merging**